packages/@aws-cdk/aws-glue-alpha/lib/table.ts

```diff
@@ -421,7 +455,8 @@ export class Table extends Resource implements ITable {
    *
    * @param grantee the principal
    */
-  public grantRead(grantee: iam.IGrantable): iam.Grant {
+  public grantRead(grantee: iam.IGrantable): iam.Grant | undefined {
+    if (!this.bucket) return;
```

Comment:
This is I don't think this is accurate. you probably only want to skip the S3 stuff here.

Reply:
I see what you mean, there's a lot of functionality missing for this. I'll get to work on it

Reply:
This has been updated